tests/src/common/test_case/builder.rs

```diff
@@ -272,10 +269,13 @@ where
             &mut test_case.app,
             test_case.address_book.lease_code(),
             test_case.address_book.lpp().clone(),
-            test_case.address_book.time_alarms().clone(),
-            test_case.address_book.oracle().clone(),
+            (
+                test_case.address_book.time_alarms().clone(),
+                test_case.address_book.oracle().clone(),
+            ),
             test_case.address_book.profit().clone(),
             test_case.address_book.reserve().clone(),
+            test_case.address_book.protocols_registry().clone(),
```

Comment:
If we need so many parameters, it might be better to introduce a `struct` containing all addresses, rather than packing them in a tuple. At the very least it's really hard mix them up.